alot/commands/globals.py

```diff
@@ -396,16 +396,18 @@ def __init__(self, command, **kwargs):
         Command.__init__(self, **kwargs)
         self.command = command
 
-    def apply(self, ui):
+    async def apply(self, ui):
         try:
             hooks = settings.hooks
             if hooks:
                 env = {'ui': ui, 'settings': settings}
                 for k, v in env.items():
-                    if k not in hooks.__dict__:
-                        hooks.__dict__[k] = v
+                    if not getattr(hooks, k, None):
+                        setattr(hooks, k, v)
 
-            exec(self.command)
+            t = eval(self.command)
```

Comment:
This cannot be fixed, and I don't think that trading `exec` for `eval` is the end of the world. Code being executed is arbitrary anyway, since it's provided by a module that the user provides.

I guess we could do something like add a "call_hook" command that acts like a call command, but loads a name from the hooks file and does some sort of argument forwarding. But that's a more aggressive change.

Reply:
agreed, let's keep things simple here